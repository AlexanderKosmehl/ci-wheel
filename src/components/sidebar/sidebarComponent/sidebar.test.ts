import {
  describe, expect, it, vi,
} from 'vitest';
import generateSidebar from './sidebar';

describe('generateSidebar', () => {
  const listChangeCallback = vi.fn((_updatedEntries: string[]) => {});
  const openImportCallback = vi.fn(() => {});
  const openArchiveCallback = vi.fn(() => {});
  const listEntries = ['Test 1', 'Test 2'];

  const sidebar = generateSidebar({
    listEntries,
    listChangeCallback,
    openImportModal: openImportCallback,
    openArchiveModal: openArchiveCallback,
  });

  const buttons = Array.from(sidebar.querySelectorAll('button'));

  it('generates component correctly', () => {
    expect(sidebar).toMatchSnapshot();
  });

  it('adds new entries correctly', () => {
    const input = sidebar.querySelector<HTMLInputElement>('input');
    const inputButton = sidebar.querySelector<HTMLButtonElement>('.addButton');

    if (!input || !inputButton) throw Error('Input not rendered correctly!');

    const newEntry = 'Test 3';
    input.value = newEntry;
    input.dispatchEvent(new Event('keyup'));

    inputButton.click();

    expect(listChangeCallback).toBeCalled();
    expect(listChangeCallback).toBeCalledWith([newEntry, ...listEntries]);
  });

  it('binds import callback correctly', () => {
    const importButton = buttons.find((button) => button.textContent === 'Import');

    if (!importButton) throw Error('No input button found!');

    importButton.click();
    expect(openImportCallback).toBeCalled();
  });

  it('binds archive callback correctly', () => {
    const archiveButton = buttons.find((button) => button.textContent === 'Archiv');

    if (!archiveButton) throw Error('No archive button found!');

    archiveButton.click();
    expect(openArchiveCallback).toBeCalled();
  });
});
