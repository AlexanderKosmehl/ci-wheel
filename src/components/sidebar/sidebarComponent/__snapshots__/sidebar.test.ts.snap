// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateSidebar generates component correctly 1`] = `
<div
  class="sidebar"
  hidden=""
  id="sidebar"
>
  <div
    class="container"
  >
    <input
      class="input"
      placeholder="Neuer Eintrag"
      type="text"
    />
    <button
      class="button addButton"
      disabled=""
    >
      <img
        class="icon"
        src="test-file-stub"
      />
    </button>
  </div>
  <ul
    class="listContainer"
  >
    <li
      class="container"
    >
      <span
        class="label"
      >
        Test 1
      </span>
      <button
        class="button deleteButton"
      >
        <img
          class="icon"
          src="test-file-stub"
        />
      </button>
    </li>
    <li
      class="container"
    >
      <span
        class="label"
      >
        Test 2
      </span>
      <button
        class="button deleteButton"
      >
        <img
          class="icon"
          src="test-file-stub"
        />
      </button>
    </li>
  </ul>
  <div
    class="container"
  >
    <button
      class="footerButton"
    >
      Import
    </button>
    <button
      class="footerButton"
    >
      Archiv
    </button>
  </div>
</div>
`;
